src/core/videosource.cpp

```diff
@@ -633,12 +643,19 @@ bool FFMS_VideoSource::HasPendingDelayedFrames() {
 
 bool FFMS_VideoSource::DecodePacket(AVPacket *Packet) {
     std::swap(DecodeFrame, LastDecodedFrame);
-    avcodec_send_packet(CodecContext, Packet);
+    ResendPacket = false;
 
-    int Ret = avcodec_receive_frame(CodecContext, DecodeFrame);
+    int Ret = avcodec_send_packet(CodecContext, Packet);
+    if (Ret == AVERROR(EAGAIN)) {
+        // Send queue is full, so stash packet to resend on the next call.
+        DelayCounter--;
```

Comment:
If we can delay more than once, shouldn't the stashed packet be a stash queue?

Reply:
No, since we don't read another packet until the stashed one is successfully sent.